lapce-app/src/doc.rs

```diff
@@ -783,6 +809,25 @@ impl Backend for DocBackend {
         doc.retrieve_head();
     }
 
+    fn init_parser(doc: &Document<Self>) {
```

Comment:
Just learning rust but could you do something like this?
```rs
let code = doc.buffer.get_untracked().to_string();

doc.backend.syntax.with_untracked(|syntax| {
    let syntax_styles = syntax.styles.is_some();

    let syntax_value = syntax_styles.then(|| syntax.clone());

    doc.backend.parser.borrow_mut().update_code(
        code,
        &doc.buffer.get_untracked(),
        syntax_value,
    );
});
```

Reply:
This makes code harder to understand without inlay hints or knowing all libstd functions